pkgs/applications/editors/vscode/vscode.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv, lib, callPackage, fetchurl, fetchpatch, ... } @ args:
+
+callPackage ./generic.nix (args // rec {
+
+  # How can I get plat and archive_fmt from generic.nix?
+  #inherit plat archive_fmt;
+
+  plat = {
+    "i686-linux" = "linux-ia32";
+    "x86_64-linux" = "linux-x64";
+    "x86_64-darwin" = "darwin";
+  }.${system};
+
+  archive_fmt = if system == "x86_64-darwin" then "zip" else "tar.gz";
+
+  inherit (stdenv.hostPlatform) system;
```

Comment:
`plat`, `archive_fmt`, `system` and `sha256` are not used arguments for a call to generic.nix, move them to a `let ... in`.

Reply:
Like this?

```nix
let
  inherit (stdenv.hostPlatform) system;

  plat = {
    "i686-linux" = "linux-ia32";
    "x86_64-linux" = "linux-x64";
    "x86_64-darwin" = "darwin";
  }.${system};

  archive_fmt = if system == "x86_64-darwin" then "zip" else "tar.gz";

  sha256 = {
    "i686-linux" = "0n2k134yx0zirddi5xig4zihn73s8xiga11pwk90f01wp1i0hygg";
    "x86_64-linux" = "0ljijcqfyrfck5imldis3hx9d9iacnspgnm58kvlziam8y0imwzv";
    "x86_64-darwin" = "00fg106rggsbng90k1jjp1c6nmnwni5s0fgmbz6k45shfa3iqamc";
  }.${system};
in
  callPackage ./generic.nix (args // rec {
...
```

Reply:
Yes. Can drop `rec` probably.